### PR TITLE
Add users to reviews

### PR DESCRIPTION
**Pull request by tjswierzewski**
Add user ownership for reviews and prevent users who are not signed in form making a review

### CODE REVIEW

File: app/javascript/react/components/CoffeeShopShowContainer.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
-import React, { useState, useEffect, Fragment } from "react";
-import ReviewsContainer from "./ReviewsContainer";
-import ReviewFormContainer from "./ReviewFormContainer";
+import React, { useState, useEffect, Fragment } from 'react';
+import ReviewsContainer from './ReviewsContainer';
+import ReviewFormContainer from './ReviewFormContainer';
+import ErrorList from './ErrorList';
```
Comment (dovakeen118) [inline app/javascript/react/components/CoffeeShopShowContainer.js]: doesn't look like this ErrorList component is used in this file?

File: app/models/user.rb [hunk 2]
```diff
@@ -4,6 +4,8 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
 
+  has_many :reivews
```
Comment (dovakeen118) [inline app/models/user.rb]: does a user also have many coffee shops?

File: app/javascript/react/components/CoffeeShopShowContainer.js [hunk 3]
```diff
@@ -28,21 +30,25 @@ const CoffeeShopShowContainer = (props) => {
...
       if (reviewResponse.ok) {
         const parsedReviewResponse = await reviewResponse.json();
 
         setCoffeeShop({
           ...coffeeShop,
-          reviews: [...coffeeShop.reviews, parsedReviewResponse.review],
+          reviews: [...coffeeShop.reviews, parsedReviewResponse.review]
         });
       }
+      if (reviewResponse.status === 401) {
```
Comment (dovakeen118) [inline app/javascript/react/components/CoffeeShopShowContainer.js]: Looks like your controller is set up to send an error of unprocessable entity, which is a 422 error that isn't being checked for here. Additionally, the authenticate_user implementation will send a 400 status code but it won't send an error body with a message - you have the option to look for this status code and hard coding a message that the user must be logged in, or defining your own authenticate_user method in the API controller (the built in Devise authenticate user method isn't set up to work with a controller that returns JSON)

`authenticate_user!` essentially checks if there is a `current_user` in session or not, but won't provide a message. You could create your own reusable method in the `application_controller`, something like
```
  def authenticate_user
    if !user_signed_in?
      render json: {error: ["You need to be signed in first"]}
    end
  end
```

File: app/javascript/react/components/CoffeeShopShowContainer.js [hunk 4]
```diff
@@ -67,7 +73,7 @@ const CoffeeShopShowContainer = (props) => {
       <p>{coffeeShop.description}</p>
       <div>
         <h2>Reviews</h2>
-        <ReviewFormContainer addNewReview={addNewReview} />
+        <ReviewFormContainer addNewReview={addNewReview} setErrors={setErrors} errors={errors} />
```
Comment (dovakeen118) [inline app/javascript/react/components/CoffeeShopShowContainer.js]: up to you, but you can potentially conditionally render the form whether of not the current user is signed in. In that case, you would need to track the current user in state which could be serialized in your SHOW fetch request

**Review by dovakeen118 — COMMENTED** [review]
Some comments on your fetch request error handling!

File: app/javascript/react/components/CoffeeShopShowContainer.js (hunk 3 above)
Comment (tjswierzewski) [reply app/javascript/react/components/CoffeeShopShowContainer.js]: The Devise method give an error message in the returned JSON

File: app/models/user.rb [hunk 5]
```diff
@@ -3,8 +3,12 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
-  mount_uploader :profile_photo, ProfilePhotoUploader
-  
+
+  has_many :reviews
+  has_many :coffee_shops
+
+  # mount_uploader :profile_photo, ProfilePhotoUploader
```
Comment (nwalberts) [inline app/models/user.rb:10]: remove this if the feature isnt working

**Pull request merged**